src/rgw/setup/rgw.py

```diff
@@ -147,17 +158,19 @@ def start(conf: MappedConf, index: str):
         motr_fid_value = Conf.get(Rgw._conf_idx, motr_fid_key)
         log_path = Rgw._get_log_dir_path(conf)
         motr_trace_dir = os.path.join(log_path, 'motr_trace_files')
-        addb_dir = os.path.join(log_path, f'addb_files-{motr_fid_value}')
         os.makedirs(motr_trace_dir, exist_ok=True)
+
+        addb_dir = os.path.join(log_path, f'addb_files-{motr_fid_value}')
         os.makedirs(addb_dir, exist_ok=True)
+
         # Create rgw crash file directory
         rgw_core_dir = os.path.join(log_path, 'rgw_debug')
         os.makedirs(rgw_core_dir, exist_ok=True)
 
         Log.info('Starting radosgw service.')
         log_file = os.path.join(log_path, f'{const.COMPONENT_NAME}_startup.log')
 
-        RgwService.start(conf, config_file, log_file, motr_trace_dir, rgw_core_dir, index)
+        RgwService.start(conf, config_file, log_file, motr_trace_dir, addb_dir, rgw_core_dir, index)
```

Comment:
argument list here keeps growing.
@sachitanands we need to change all these inputs into a dictionary and take that as i/p in future, keep an improvment for later

Reply:
okay.. sure. will create ticket for this.